Assets/Scripts/settingsManager.cs

```diff
@@ -0,0 +1,73 @@
+using UnityEngine;
+using System.IO; // Used to read/write files for saving settings
+
+// This class manages saving, loading, and accessing player settings (singleton pattern)
+public class SettingsManager : MonoBehaviour
+{
+    // Singleton instance to allow global access to settings
+    public static SettingsManager Instance;
+
+    // Stores all the actual setting values
+    public SettingsData settings;
+
+    // File path where the settings JSON will be saved
+    private string settingsPath;
+
+    // Called when the script is first loaded (before Start)
+    void Awake()
+    {
+        // Ensure only one SettingsManager exists
+        if (Instance == null)
+        {
+            Instance = this;
+            DontDestroyOnLoad(gameObject); // Keep this object alive between scene changes
+
+            // Define the file path for saving the settings JSON file
+            settingsPath = Application.persistentDataPath + "/settings.json";
```

Comment:
Given that Application.presistentDataPath is a static variable, the settings path can also be a static variable since we don't expect it to change

Reply:
Never mind. Apparently Application.persistentDataPath needs to be done on Awake.